source/MaterialXCore/Types.h

```diff
@@ -426,6 +470,42 @@ template <class M, class S, size_t N> class MatrixN : public MatrixBase
         return getAdjugate() / getDeterminant();
     }
 
+    /// @}
+    /// @name 3D Transformations
+    /// @{
+
+    /// Set matrix to identity
+    void setIdentity()
```

Comment:
It might be more efficient to encourage the client to use the IDENTITY constants instead of calling a "set" method on an existing matrix:

```
Matrix44 mat(Matrix44::IDENTITY)
```

Using an IDENTITY constant, the client is less likely to write code that overwrites their matrix multiple times, and there shouldn't be any unnecessary copies (other than the single copy from the constant IDENTITY matrix to the newly constructed, uninitialized matrix).